### PR TITLE
Remove broker.spec.config defaulting magic

### PR DESCRIPTION
**Pull request by pierDipi**
## Proposed Changes

- Remove broker.spec.config defaulting magic
- Add webhook validation (i think we're better off rejecting invalid Brokers from the start)

**Release Note**

```release-note
`broker.spec.config` is now required.
```

**Docs**

- I'll check if docs mentions this feature (I don't thing it does)

/assign @aliok

### CODE REVIEW

**Comment by matzew** [comment]
> `broker.spec.config` is now required.

Isn't that quiet some implication? Or are we generally sure our users do specifiy that ? And don't default on any "defaulting" ?

**Comment by pierDipi** [comment]
> > `broker.spec.config` is now required.
> 
> Isn't that quiet some implication? Or are we generally sure our users do specifiy that ? And don't default on any "defaulting" ?

I'm removing the magic defaulting meaning that if users don't specify any `config` and there is no defaulting provided by the eventing (core) webhook (cluster or namespace defaults), then it will fail.

**Comment by aliok** [comment]
> Isn't that quiet some implication? Or are we generally sure our users do specifiy that ? And don't default on any "defaulting" ?

There was a mechanism that auto-creates the broker, using the defaults in the configmap. Is this still the case? Do we care?

File: control-plane/pkg/reconciler/broker/broker_test.go [hunk 1]
```diff
@@ -121,6 +118,7 @@ func brokerReconciliation(t *testing.T, format string, env config.Env) {
 			Name: "Reconciled normal - no DLS",
 			Objects: []runtime.Object{
 				NewBroker(),
+				BrokerConfig(bootstrapServers, 20, 5),
```
Comment (aliok) [inline control-plane/pkg/reconciler/broker/broker_test.go:121]: I like this better than my approach

**Comment by pierDipi** [comment]
This feature will still provide the defaulting mechanism https://knative.dev/docs/eventing/broker/kafka-broker/#set-as-default-broker-implementation

File: control-plane/pkg/reconciler/broker/broker_test.go (hunk 1 above)
Comment (aliok) [reply control-plane/pkg/reconciler/broker/broker_test.go:121]: Can you remove these though?

```
OtherTestData: map[string]interface{}{
				kafka.BootstrapServersConfigMapKey: bootstrapServers,
			},
```

File: control-plane/pkg/reconciler/broker/broker_test.go [hunk 2]
```diff
@@ -1970,6 +1995,7 @@ func brokerFinalization(t *testing.T, format string, env config.Env) {
 			Name: "Reconciled normal - no broker found in config map",
 			Objects: []runtime.Object{
 				NewDeletedBroker(),
+				BrokerConfig(bootstrapServers, 20, 5),
```
Comment (aliok) [inline control-plane/pkg/reconciler/broker/broker_test.go:1998]: Are there any tests that make sure the config is actually read from the configmap?

File: control-plane/pkg/reconciler/broker/broker_test.go (hunk 1 above)
Comment (pierDipi) [reply control-plane/pkg/reconciler/broker/broker_test.go:121]: Yes, we can remove that.

File: control-plane/pkg/reconciler/broker/broker_test.go (hunk 2 above)
Comment (pierDipi) [reply control-plane/pkg/reconciler/broker/broker_test.go:1998]: Yes, all of them, and there is a test that fails because we can't find the ConfigMap.

**Comment by aliok** [comment]
lgtm

Leaving final approval to @matzew

**Pull request merged**